sdk/src/main/java/com/deploygate/sdk/DeployGateClient.java

```diff
@@ -0,0 +1,138 @@
+package com.deploygate.sdk;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.content.pm.Signature;
+import android.content.pm.SigningInfo;
+import android.os.Build;
+
+import com.deploygate.sdk.internal.Logger;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+class DeployGateClient {
+    private static final String[] FINGERPRINTS = new String[]{
+            // deploygate release
+            "2f97f647645cb762bf5fc1445599a954e6ad76e7",
+            // mba debug
+            "c1f285f69cc02a397135ed182aa79af53d5d20a1"
+    };
+
+    public final long versionCode;
+    public final boolean isInstalled;
+    private final long featuresFlag;
+
+    @SuppressLint("PackageManagerGetSignatures")
+    DeployGateClient(
+            Context context,
+            String packageName
+    ) {
+        PackageInfo info = null;
+        int flag = PackageManager.GET_META_DATA;
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            flag |= PackageManager.GET_SIGNING_CERTIFICATES;
+        } else {
+            flag |= PackageManager.GET_SIGNATURES;
+        }
+
+        try {
+            info = context.getPackageManager().getPackageInfo(packageName, flag);
+        } catch (PackageManager.NameNotFoundException e) {
+            Logger.w("deploygate app is not found");
+        }
+
+        if (info != null) {
+            this.isInstalled = checkSignature(info);
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                this.versionCode = info.getLongVersionCode();
+            } else {
+                this.versionCode = info.versionCode;
+            }
+
+            this.featuresFlag = info.applicationInfo.metaData.getLong("com.deploygate.features", 0);
```

Comment:
The client app will tell SDK about supported features through metadata. (bitmasks)